src/s3path.jl

```diff
@@ -118,14 +123,21 @@ function Base.tryparse(::Type{<:S3Path}, str::AbstractString; config::Union{Noth
         path = Tuple(filter!(!isempty, tokenized[4:end]))
     end
 
-    return S3Path(path, root, drive, isdirectory, config, nothing)
+    version = length(version_split) == 1 ? nothing : version_split[2]
+    return S3Path(path, root, drive, isdirectory, config, version)
 end
 
 function normalize_bucket_name(bucket)
     return strip(startswith(bucket, "s3://") ? bucket : "s3://$bucket", '/')
 end
 
-Base.print(io::IO, fp::S3Path) = print(io, fp.anchor * fp.key)
+function Base.print(io::IO, fp::S3Path)
+    str = fp.anchor * fp.key
+    if fp.version !== nothing && !isempty(fp.version)
+        str *= "?versionId=" * fp.version
+    end
+    return print(io, str)
```

Comment:
Avoid creating intermediate strings
```suggestion
    print(io, fp.anchor, fp.key)
    if fp.version !== nothing && !isempty(fp.version)
        print(io, "?versionId=", fp.version)
    end
    return nothing
end
```

Reply:
Done!